.github/workflows/test-deploy.yaml

```diff
@@ -30,32 +41,60 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
 
+      - name: Set UDS CLI Arch
+        id: setUdsCliArch
+        shell: bash
+        run: INPUT=${{ runner.arch == 'X64' && 'amd64' || runner.arch }}; echo "ARCH=${INPUT,,}" >> "$GITHUB_OUTPUT"
+
+      - name: Install UDS CLI
+        shell: bash
+        run: curl -o /usr/local/bin/uds -L https://github.com/defenseunicorns/uds-cli/releases/download/v${{ inputs.udsCliVersion }}/uds-cli_v${{ inputs.udsCliVersion }}_Linux_${{steps.setUdsCliArch.outputs.ARCH}} && chmod +x /usr/local/bin/uds
```

Comment:
this should probably be an external GH action

Reply:
yeah understood per our earlier discussion.

Reply:
This is waiting on the action repo i created yesterday to be made public